exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -413,7 +413,7 @@ func (r *MachinePoolReconciler) createOrUpdateMachines(ctx context.Context, mp *
 			log.Info("Creating new Machine for infraMachine", "infraMachine", klog.KObj(infraMachine))
 			machine := computeDesiredMachine(mp, infraMachine, nil)
 
-			if err := r.Client.Create(ctx, machine); err != nil {
+			if err := ssa.Patch(ctx, r.Client, MachinePoolControllerName, machine); err != nil {
```

Comment:
this applies @sbueringer's suggestion from https://github.com/kubernetes-sigs/cluster-api/pull/9619#discussion_r1395699740

Reply:
I think the issue here is that we are using a fake client which does not properly support server side apply in this case.

It returns here: https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.3/pkg/client/fake/client.go#L828

Reply:
@chrischdi Is right, fake client doesn't support SSA. Sorry didn't think about that when recommending using SSA.

For reference. In KCP we previously had a flag to skip some optional SSA part until we migrated the tests over to SSA (https://github.com/kubernetes-sigs/cluster-api/issues/8393).

But I think we should aim for using SSA consistently before the v1.6.0 release. Otherwise we end up with strange issues in our release. 

The ideal solution to this is to migrate the affected tests to envtest.

Reply:
As the PR is otherwise green we could also move this part to a separate PR if you prefer

Reply:
removed for now, let's do a separate PR

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/9755